pkg/route/validation/validation_test.go

```diff
@@ -1800,6 +1867,13 @@ func TestValidateTLS(t *testing.T) {
 		name           string
 		route          *routev1.Route
 		expectedErrors int
+
+		// fields for externalCertificate
+		allow     bool
+		validType bool
+		validNS   bool
```

Comment:
Why obfuscating the test case structure by adding these indirections which are used to build a `Secret`? Why not pass a `Secret` as a whole, similar to `Route`?

Reply:
Thanks for the suggestion. I've updated the structure of the testCases

Reply:
I still see `allow` `validType` and `validNS` fields.